### ISSUE TITLE
Under WSL, `spt` does nothing on invocation

### ISSUE THREAD

**Issue by ix** [Rigellute/spotify-tui#182]
I have installed spotify-tui both through `cargo install spotify-tui` and by building from git master, in both cases running `spt` results in nothing happening and the program exiting.

```
spotify-tui $ cargo run
    Finished dev [unoptimized + debuginfo] target(s) in 0.20s
     Running `target/debug/spt`
spotify-tui $
```

On the first run I was prompted for my credentials, which I entered. I have tried deleting `~/.config/spotify-tui` and re-authenticating to see if that fixed it, which it did not.

Let me know if any further information is needed!

Cargo/Rust versions:
```
spotify-tui $ cargo --version
cargo 1.39.0 (1c6ec66d5 2019-09-30)
spotify-tui $ rustc --version
rustc 1.39.0 (4560ea788 2019-11-04)
```

**Comment by ix** [comment]
Some additional information:
After entering the Client ID and Client Secret on the first run or after deleting the config directory, the following message is displayed in the program output

```
Error Os { code: 2, kind: NotFound, message: "No such file or directory" };Please navigate here ["https://accounts.spotify.com/authorize?[some redacted params]"]
```

**Comment by Rigellute** [comment]
Sorry to hear this @ix, which operating system are you on?

**Comment by ix** [comment]
Hi there Rigellute, thanks for a timely response.

/etc/lsb-release lists the following;
```
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=18.04
DISTRIB_CODENAME=bionic
DISTRIB_DESCRIPTION="Ubuntu 18.04.3 LTS"
```

This is in Windows Subsystem for Linux, with a full `uname` of 
```
Linux ahostname 4.4.0-18362-Microsoft #476-Microsoft Fri Nov 01 16:53:00 PST 2019 x86_64 x86_64 x86_64 GNU/Linux
```

Windows specifications are as follows:
![image](https://user-images.githubusercontent.com/1900521/69981405-28783e00-152a-11ea-9299-b5b4254b2e7c.png)

I have tried running the program both in and out of tmux to no avail. `xorg-dev` is installed, alongside the libxcb* dependencies... `xorg-dev is already the newest version (1:7.7+19ubuntu7.1)`

I am running the program in a headless environment, perhaps that could be related to it as I assume an X server is needed for the X clipboard functionality to work? I will test later with X11 support through Xming if needed.

**Comment by Rigellute** [comment]
Hmm... thanks for this extra information.

If it is not too much bother, would you mind reproducing this in dev mode? i.e. clone the repo and `cargo run`?

That might give you the error causing the app to exit immediately.

**Comment by ix** [comment]
I've worked it out! 

With Xming (a Windows X server) running, invoking `spt` like follows causes it to work fine:
```env DISPLAY=localhost:0.0 ~/.cargo/bin/spt```

Removing the environment variable with a valid X11 display causes the "silent fail" behavior I initially reported to occur. Since you mentioned X support being a requirement for the clipboard functionality, would it be possible to make this able to be disabled through a flag?

I would be more than happy to contribute this change if desirable, but it may take me a week or so to have the time for it.

**Comment by ix** [comment]
Just to add to the above: `xfvb-run spt` also works, by running spt in the context of an in-memory X Server, and is useful when there isn't actually an X server set up to connect to.

**Comment by Rigellute** [comment]
Great that you've figured it out! And thanks for the write up.

You are right, we should probably gracefully handle the error when the clipboard can't be set up. This should be easy to implement. Will investigate setting up a flag to toggle the clipboard behaviour.

**Comment by Rigellute** [comment]
This was fixed in #217
